src/components/StyledCards.js

```diff
@@ -26,5 +26,10 @@ const StyledCard = styled.div`
 		css`
 			background-color: #f68787;
 		`}
+		${({variant = 'sets'}) =>
+		variant === 'sets' &&
+		css`
+			background-color: #cfcfc9;
+		`}
```

Comment:
Here you only destructure `variant` from your props object, you do not need to add a value:

```suggestion
		${({variant}) =>
		variant === 'sets' &&
		css`
			background-color: #cfcfc9;
		`}
```

As you have quite a lot of different variants here, I would suggest to do something similar as in this [CodeSandbox](https://codesandbox.io/s/bo-web-22-1-styled-components-vs-bem-8083tx?file=/src/components/StyledButton.js). Have a look at the different variants of `padding` in the `sizeMap` object.

Reply:
i dont understand how this makes sense to my case, i tried to implement it and at the end it feels like i have more code, could you give me a better example of my case please? I am happy to improve

Reply:
![Screenshot 2022-08-25 at 11 19 03](https://user-images.githubusercontent.com/102238180/186626794-54923584-dbd8-487c-8ca2-1024aae19b29.png)

sollte ich das einfach so umsetzten? Oder wolltest du etwas anderes von mir? Bin etwas verwirrt, vllt denke ich zu kompliziert

Reply:
Have a look at the CodeSandbox again. What is done there for the `padding`, you can do for the `background-color`. Meaning: create an object with the different variants (same as `sizeMap` object). Then you need only one line in your styled component to set the `background-color` depending on the variant - similar to the following in the example:

```
 padding: ${({ size = 'default' }) => {
    return sizeMap[size];
  }};
```

Reply:
thank you got it!